pkg/prometheus/agent/operator.go

```diff
@@ -564,6 +581,84 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 
 	p := pobj.(*monitoringv1alpha1.PrometheusAgent)
 	p = p.DeepCopy()
+	if p.Spec.Mode != nil && *p.Spec.Mode == "DaemonSet" {
+		err = c.syncDaemonSet(ctx, key, p)
+		c.reconciliations.SetStatus(key, err)
+		return err
+	}
+	err = c.syncStatefulSet(ctx, key, p)
+	c.reconciliations.SetStatus(key, err)
+	return err
+}
+
+func (c *Operator) syncDaemonSet(ctx context.Context, key string, p *monitoringv1alpha1.PrometheusAgent) error {
```

Comment:
the first thing to verify should be that the feature gate is enabled and return an error if not.